languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -36090,19 +36090,57 @@ though, and has since been slightly modified)-->
       <example correction=","><marker>dans ce cas</marker> il est certain</example>
       <example><marker>dans ce cas</marker>, il est certain</example>
     </rule>
-    <rule default="off" id="ET_CE" name="et ce (sans virgule après)">
-      <pattern>
-        <marker>
-          <token>et</token>
-          <token>ce</token>
-        </marker>
-        <token>
-          <exception regexp="yes">[,.;!:\-?n]</exception></token>
-      </pattern>
-      <message>« Et ce » (sans virgule après) peut être une faute de typographie. Employez une virgule (<suggestion>,</suggestion>) au besoin.</message>
-      <example correction=","><marker>et ce</marker> blah blah</example>
-      <example><marker>et ce</marker>,</example>
-    </rule>
+      <rule default="off" id="VIRGULE_ET_CE" name="virgule et ce">
+          <antipattern>
+              <token regexp="yes">[^,‘’–)\[—."\/;\]\|:(\…'«»“”-]+</token>
+              <token>et</token>
+              <token>ce</token>
+              <token postag="SENT_END"/>
+          </antipattern>
+          <antipattern>
+              <token postag="[ACPD].*" postag_regexp="yes"/>
+              <token>et</token>
+              <token>ce</token>
+              <token postag="[PA].*" postag_regexp="yes"/>
+          </antipattern>
+          <antipattern>
+              <token regexp="yes">[^,‘’–)\[—."\/;\]\|:(\…'«»“”-]+</token>
+              <token>et</token>
+              <token>ce</token>
+              <token postag="[PA].*" postag_regexp="yes"/>
+              <token postag="SENT_END"/>
+          </antipattern>
+          <antipattern>
+              <token regexp="yes">[^,‘’–)\[—."\/;\]\|:(\…'«»“”-]+</token>
+              <token>et</token>
+              <token>ce</token>
+              <token regexp="yes">ne|n(&apostrophe;)|passage|n</token>
+          </antipattern>
+          <antipattern>
+              <token regexp="yes">[^,‘’–)\[—."\/;\]\|:(\…'«»“”-]+</token>
+              <token>et</token>
+              <token>ce</token>
+              <token postag="[PA].*" postag_regexp="yes" min="0" max="1"/>
+              <token regexp="yes">pourquoi|quoi|qu(&apostrophe;)</token>
+          </antipattern>
+          <pattern>
+              <marker>
+                  <token regexp="yes">[^,‘’–)\[—."\/;\]\|:(\…'«»“”-]+<exception postag="[PD].*" postag_regexp="yes"/></token>
+                  <token case_sensitive="yes">et</token>
+                  <token  case_sensitive="yes">ce</token>
+              </marker>
+              <token postag="[PA].*" postag_regexp="yes"><exception regexp="yes">qu[ie]?</exception></token>
+          </pattern>
+          <message>Des virgules sont conseillées dans ce contexte.</message>
```

Comment:
This is a cool rule, but a URL would be great.
Also, can you add an AP to PAS_DE_VIRGULE for "et ce"? If I'm not mistaken, there might still be some cases where that rule will try and erase the comma.
You can make that change to PAS_DE_VIRGULE in this same pull request.

Reply:
I think a subrule would be in order for such cases where the user puts a comma before "et ce" but not after, and for when the user puts a comma after "et ce", but not before.

Reply:
I didn't see any cases where the rule would erase the comma as I already put an exception in the first subrule. 
I added the subrules as required.